### DISCUSSION TITLE
How to use pixi to achieve motion trajectory?  Who can teach me

### DISCUSSION THREAD

**Discussion by bibiehy** [pixijs/pixijs#8572] [Q&A]
pixi@6.4.2

I am a beginner, which teacher can help me.
Don't know how to do it, sorry I didn't provide an example on codepen.io.

I want to achieve such an interaction, if there are five people, they have their own movement trajectory coordinates, each person's data looks like this [{x, y, status: 'walking'}, {x, y, status: 'riding '}], when I press the space bar, they draw their own trajectories on the map, and need to dynamically change the state of the person (sprite), the state is walking and riding.

Can anyone help me, please, preferably with an example at codepen.io, i will be very grateful.

Similar to this effect,but this is svg

![GIF](https://user-images.githubusercontent.com/35096473/183720404-37a5b94d-bc41-464b-b94c-14c98c42f793.gif)

**Comment by domis86** [comment]
This problem isnt really about PixiJS (PixiJS is just a tool to draw things).

I will try to explain general directions which you should explore to achieve your goal:
* _"they have their own movement trajectory coordinates"_
    * Probably you dont mean https://en.wikipedia.org/wiki/Trajectory because that involves things like mass and gravity (like throwing a ball etc) - and as such would be too complex for beginner (as you described yourself).
    * probably you mean things related to math / physics like: linear interpolation, velocity (constant velocity), maybe also vectors. About these see "Essential Mathematics For Aspiring Game Developers" link in bottom
* _each person's data looks like this [{x, y, status: 'walking'}, {x, y, status: 'riding '}_
    * to simplify this i would suggest to use pair of coordinates for each line (line between two points) - something like:
```
pathsOfPerson = [
{startPosition: [x, y], targetPosition: [x, y], state: 'walking'},
{startPosition: [x, y], targetPosition: [x, y], state: 'riding'},
,...
```
^ this way we see that some person goes first from `startingPosition` to `targetPosition` using "walking" state, and then from other `startingPosition` to `targetPosition` using "riding" state etc.
* _they draw their own trajectories on the map, and need to dynamically change the state of the person (sprite), the state is walking and riding._
    * you can achieve that by picking elements from `pathsOfPerson` array and drawing straight line from `startingPosition` to a position calculated using linear interpolation between `startingPosition` and `targetPosition` for a certain time (aka: duration). Said time could be calculated using following formula: `time = distance / velocity`
        * where `distance` is distance (in pixels or whatever units) between `startingPosition` and `targetPosition`. See: https://www.calculatorsoup.com/calculators/geometry-plane/distance-two-points.php
        * and `velocity` is depending on `state`. If state is "walking" then it can be 20 pixels per second, and when "riding" it can be 100 etc.
        * for linear interpolation you can create your own function (should be easy to find something on the web) or look at libs linked in bottom here.
        * Pixi Graphics tip: on each frame use https://pixijs.download/dev/docs/PIXI.Graphics.html#clear and then draw partial line as described above. See how is used for example in: https://pixijs.io/examples/#/graphics/dynamic.js
    * when line from one element of `pathsOfPerson` is completed then leave it drawn on screen, pick next element and start drawing again as described above (create new PIXI Graphics object).


Animation related resources / libs which you can use and/or learn from:
* "Essential Mathematics For Aspiring Game Developers" by javidx9: https://www.youtube.com/watch?v=DPfxjQ6sqrc - "This video outlines what I believe are some of the core principles you need to understand to make dynamic computer games, covering vectors, angles and motion. " (includes linear interpolation).
* https://github.com/miltoncandelero/tweedle.js  + https://www.pixijselementals.com/#tweens
* https://github.com/tweenjs/tween.js/
* https://github.com/davidfig/pixi-ease
* https://animejs.com/documentation/#JSobject + https://animejs.com/documentation/#linearEasing + https://animejs.com/documentation/#duration

**Reply by bibiehy** [reply]
Thank you for your patience, let me take a look.
